momento-redis/list_commands.go

```diff
@@ -4,145 +4,235 @@ import (
+func (m *MomentoRedisClient) LMPop(ctx context.Context, direction string, count int64, keys ...string) *redis.KeyValuesCmd {
 
 	panic(UnsupportedOperationError("This operation has not been implemented yet"))
 }
 
-func (m *MomentoRedisClient) LPop(ctx context.Context, key string) *StringCmd {
+func (m *MomentoRedisClient) LPop(ctx context.Context, key string) *redis.StringCmd {
 
 	panic(UnsupportedOperationError("This operation has not been implemented yet"))
 }
 
-func (m *MomentoRedisClient) LPopCount(ctx context.Context, key string, count int) *StringSliceCmd {
+func (m *MomentoRedisClient) LPopCount(ctx context.Context, key string, count int) *redis.StringSliceCmd {
 
 	panic(UnsupportedOperationError("This operation has not been implemented yet"))
 }
 
-func (m *MomentoRedisClient) LPos(ctx context.Context, key string, value string, args LPosArgs) *IntCmd {
+func (m *MomentoRedisClient) LPos(ctx context.Context, key string, value string, args redis.LPosArgs) *redis.IntCmd {
 
 	panic(UnsupportedOperationError("This operation has not been implemented yet"))
 }
 
-func (m *MomentoRedisClient) LPosCount(ctx context.Context, key string, value string, count int64, args LPosArgs) *IntSliceCmd {
+func (m *MomentoRedisClient) LPosCount(ctx context.Context, key string, value string, count int64, args redis.LPosArgs) *redis.IntSliceCmd {
 
 	panic(UnsupportedOperationError("This operation has not been implemented yet"))
 }
 
-func (m *MomentoRedisClient) LPush(ctx context.Context, key string, values ...interface{}) *IntCmd {
+func (m *MomentoRedisClient) LPush(ctx context.Context, key string, values ...interface{}) *redis.IntCmd {
 
 	panic(UnsupportedOperationError("This operation has not been implemented yet"))
 }
 
-func (m *MomentoRedisClient) LPushX(ctx context.Context, key string, values ...interface{}) *IntCmd {
+func (m *MomentoRedisClient) LPushX(ctx context.Context, key string, values ...interface{}) *redis.IntCmd {
 
 	panic(UnsupportedOperationError("This operation has not been implemented yet"))
 }
 
-func (m *MomentoRedisClient) LRange(ctx context.Context, key string, start, stop int64) *StringSliceCmd {
+func (m *MomentoRedisClient) LRange(ctx context.Context, key string, start, stop int64) *redis.StringSliceCmd {
+	resp := &redis.StringSliceCmd{}
+	startIndex := int32(start)
+	stopIndex := int32(stop + 1) // Momento uses exclusive end, Redis uses inclusive end
+	var listLength int32
 
-	panic(UnsupportedOperationError("This operation has not been implemented yet"))
+	// Get the length of the list
+	lengthResp, err := m.client.ListLength(ctx, &momento.ListLengthRequest{
+		CacheName: m.cacheName,
+		ListName:  key,
+	})
+	if err != nil {
+		resp.SetErr(RedisError(err.Error()))
+		return resp
+	}
+	switch r := lengthResp.(type) {
+	case *responses.ListLengthHit:
+		listLength = int32(r.Length())
+	case *responses.ListLengthMiss:
+		listLength = int32(0)
+	}
```

Comment:
We only use `listLength` if either `start` or `stop` are negative, so we can save the expense of the API call by skipping this request if both are positive.